insanely_fast_whisper_rocm/audio/conversion.py

```
@@ -24,34 +30,88 @@ def ensure_wav(
+    output_path = tmp_dir / f"{original_path.stem or 'audio'}.wav"
+
+    if ffmpeg is None:
+        # Fallback to a pure-Python conversion path when FFmpeg is unavailable.
+        try:  # pragma: no cover - optional dependency
+            from pydub import AudioSegment  # type: ignore[import]
+        except ModuleNotFoundError as exc:  # pragma: no cover - handled gracefully
+            logger.error(
+                "FFmpeg is not installed and optional dependency 'pydub' is "
+                "missing; cannot convert %s to WAV.",
+                original_path,
+            )
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)